### PR TITLE
Bubble Sort

### PR DESCRIPTION
**Pull request by janis-jenny**
### This is an algorithm created with ruby simulating the sort method

We built this algorithm with:
Ruby.

### CODE REVIEW

File: buble.rb [hunk 1]
```diff
@@ -1 +1,27 @@
...
+  end
+  print arr
+  arr
+end
+
+buble_sort
+
+bubble_sort_by(%w[hi hello hey]) do |left, right|
+  left.length - right.length
+end
```
Comment (maelfosso) [inline buble.rb]: `bubble_sort_by` is not defined anywhere

File: buble.rb [hunk 2]
```diff
@@ -1 +1,27 @@
...
+      y = arr[i + 1]
+      if x > y
+        arr[i] = y
+        arr[i + 1] = x
+        a += 1
+      end
+      i += 1
+    end
+  end
+  print arr
```
Comment (maelfosso) [inline buble.rb]: Remove it please. It's not needed.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for the job done so far.
However, there are some missing requirements

You did a good job concerning the `bubble_sort` method.
However, You've forgotten to implement the `bubble_sort_by` method.
Also, read the comments left below.

Kindly add  it and submit a new code review request.

Thanks

Mael FOSSO

File: buble.rb [hunk 3]
```diff
@@ -1 +1,39 @@
...
+      x = arr[i]
+      y = arr[i + 1]
+      if x > y
+        arr[i] = y
+        arr[i + 1] = x
+        a += 1
+      end
+      i += 1
+    end
+  end
```
Comment (kobiyoyo) [inline buble.rb:18]: What the other TSE meant is that you should remove only `puts`
Now lets add `arr` back

File: buble.rb [hunk 4]
```diff
@@ -1 +1,39 @@
-puts "hello"
+def buble_sort(arr = [1, 0, 4, 3, 5, 1, 0, 1])
```
Comment (kobiyoyo) [inline buble.rb]: - [x] Remove the default arr in the argument,it should be like this
`def bubble_sort(arr)`
- [x] Correct the spelling to `bubble_sort`

File: buble.rb [hunk 5]
```diff
@@ -1 +1,39 @@
...
+      arr2[i + 1] = x
+    end
+    i += 1
+  end
+end
+
+buble_sort
+bubble_sort_by(%w[hi hello hey]) do |left, right|
+  left.length - right.length
+end
```
Comment (kobiyoyo) [inline buble.rb]: - [ ] Let's remove this debug code

File: buble.rb [hunk 6]
```diff
@@ -1 +1,39 @@
...
+  while i < arr2l
+    value = yield(arr2[i], arr2[i + 1])
+    if value.positive?
+      x = arr2[i]
+      y = arr2[i + 1]
+      arr2[i] = y
+      arr2[i + 1] = x
+    end
+    i += 1
+  end
```
Comment (kobiyoyo) [inline buble.rb:34]: - [x] Lets add `arr2`  to return the result

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General
- [ ] Check comments below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: buble.rb (hunk 5 above)
Comment (Nexch) [reply buble.rb]: can you explain your self a little bit better

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

Thank you for applying the suggested :+1: 
You can merge it now :wink: 

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**